src/odemis/gui/util/filename_pattern.py

```diff
@@ -0,0 +1,219 @@
+# -*- coding: utf-8 -*-
+"""
```

Comment:
As there are no dependencies on wxPython, nor odemis.gui and it could be used in some general case, I'd suggest to move this module to `odemis.util`. Maybe just call it `filename`, to keep the name short.